### ISSUE TITLE
Visual Studio scaffold with error 'Install the package Microsoft.VisualStudio.Web.CodeGeneration.Design and try again.'

### ISSUE THREAD

**Issue by Rick-Anderson** [dotnet/Scaffolding#1540]
.NET 5, latest release
Visual Studio, latest release
related issue:  #1393

To reproduce the problem. In VS

* Create a new Razor Pages web app (5.0)
* Add a model

```
using System;

namespace ContosoUniversity.Models
{
    public class Student
    {
        public int ID { get; set; }
        public string LastName { get; set; }
        public string FirstMidName { get; set; }
        public DateTime EnrollmentDate { get; set; }
    }
}
```

Scaffold with model and use **+** to create a context.
The scaffolder adds the following mismatched packages before VS throws an error

```
  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.4" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="5.0.4">
      <PrivateAssets>all</PrivateAssets>
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="5.0.2" />
  </ItemGroup>
```

Expected result: Scaffolded model.

Actual results:

---------------------------
Microsoft Visual Studio
---------------------------
Error

There was an error running the selected code generator:

'Install the package Microsoft.VisualStudio.Web.CodeGeneration.Design and try again.'
---------------------------
OK   
---------------------------

The preceding error results from VS not loading the  mismatched NuGet Packages on the first try:

Work around:

Run the scaffolder again. It succeeds the 2nd time.

**Comment by grenadin** [comment]
![error](https://user-images.githubusercontent.com/11970460/113186955-7ae27f80-9282-11eb-81be-d358f069cf36.png)
I have a problem with the step at the new scaffolding item.  After click + on the Data Context class and Click Add. The error will occur. Run scaffolder again. And error every time. Try another laptop that installs vs2019 same version. Error same

**Comment by grenadin** [comment]
![error](https://user-images.githubusercontent.com/11970460/113186955-7ae27f80-9282-11eb-81be-d358f069cf36.png)
I have a problem with the step at the new scaffolding item.  After click + on the Data Context class and Click Add. The error will occur. Run scaffolder again. And error every time. Try another laptop that install vs2019 same version . Error same

**Comment by Rick-Anderson** [comment]
@grenadin  unrelated to this issue. I'd review the error message carefully.

**Comment by grenadin** [comment]
Oh Sorry sir.

**Comment by jp-diegidio** [comment]
I have had the same issue today: installed VS 2019 (Community) then going through the Razor Pages with EF Core in ASP.NET Core Tutorial (*), and retrying/restarting/rebuilding wasn't enough, scaffolding just kept failing.  I have solved it by going to the Solution's Manage NuGet Packages and actually associating those packages to the project, they weren't.  Then scaffolding worked.

**Comment by vijayrkn** [comment]
This issue is fixed in the latest 6.0 version of the package.

**Comment by xperiandri** [comment]
Not fixed at all

**Comment by Usherg49** [comment]
Try installing Microsoft.EntityFrameworkCore.Relational. It worked for me.

**Comment by xperiandri** [comment]
When versions does not match (6.0.6 and 6.0.7) for example nothing helps except aligning to the version that Visual Studio installs

**Comment by cokietech** [comment]
I uninstalled and reinstalled Microsoft.VisualStudio.Web.CodeGeneration.Design NuGet package in the package manager. Doing so brought in the CodeGenerator.Design package and added Microsoft.EntityFrameworkCore.Tools. This solved my problem.

Like others, I got the error working through "Get started with Razor Pages in ASP.NET Core" tutorial. The .NET 7 version. I tried several of the above solutions to no avail. I noticed that the project file had a package reference to Microsoft.VisualStudio.Web.CodeGeneration.Design, but it didn't show up in the project packages. The package showed installed in the Nuget package manager. I uninstalled and reinstalled it in the Nuget package manager. Doing so brought in the CodeGenerator.Design package and added Microsoft.EntityFrameworkCore.Tools. This solved my problem.
